Userland/Libraries/LibIPC/Decoder.h

```diff
@@ -149,6 +165,22 @@ class Decoder {
     }
 
 private:
+    template<size_t CurrentIndex, size_t Max, typename... VariantTypes>
+    ErrorOr<void> variant_decode_helper(size_t index, Variant<VariantTypes...>& variant)
```

Comment:
:bikeshed: but the name `variant_decode_helper` seems unnecessary, we can just call it `decode_variant`.

Reply:
Well, it is a helper because it's not the main decoding function and requires some magic initial arguments.

Reply:
IMO it's just obvious that it's a "helper" and it only adds noise to slap `_helper` onto such functions (unless such a suffix is used to disambiguate functions of the same name).